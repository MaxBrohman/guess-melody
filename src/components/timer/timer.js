import React from 'react';
// eslint-disable-next-line
export const Timer = ({time}) => (
  <div className="timer__value" xmlns="http://www.w3.org/1999/xhtml">
    {// eslint-disable-next-line
    <span className="timer__mins">{time.min}</span>}
    <span className="timer__dots">:</span>
    {// eslint-disable-next-line
    <span className="timer__secs">{time.sec}</span>}
  </div>
);
