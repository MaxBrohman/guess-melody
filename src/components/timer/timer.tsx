import React from 'react';

import {TFormatTime} from '../../reducer/types';

type TTimerProps = {
  time: TFormatTime;
}

export const Timer = ({time}: TTimerProps): JSX.Element => (
  <div className="timer__value">
    <span className="timer__mins">{time.min}</span>
    <span className="timer__dots">:</span>
    <span className="timer__secs">{time.sec}</span>
  </div>
);
