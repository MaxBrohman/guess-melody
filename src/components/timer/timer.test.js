import React from 'react';
import renderer from 'react-test-renderer';

import {Timer} from './timer';

it(`Renders Timer component correctly`, () => {
  const tree = renderer
                .create(
                    <Timer time={{min: `03`, sec: `50`}} />
                )
                .toJSON();
  expect(tree).toMatchSnapshot();
});
