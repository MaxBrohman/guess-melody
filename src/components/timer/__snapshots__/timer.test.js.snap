// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Timer component correctly 1`] = `
<div
  className="timer__value"
  xmlns="http://www.w3.org/1999/xhtml"
>
  <span
    className="timer__mins"
  >
    03
  </span>
  <span
    className="timer__dots"
  >
    :
  </span>
  <span
    className="timer__secs"
  >
    50
  </span>
</div>
`;
